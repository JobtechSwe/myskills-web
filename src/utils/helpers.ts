import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
