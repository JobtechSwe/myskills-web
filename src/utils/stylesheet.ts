const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
