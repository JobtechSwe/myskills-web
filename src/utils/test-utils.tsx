import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
