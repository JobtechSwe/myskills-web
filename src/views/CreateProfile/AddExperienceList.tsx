import React from 'react'
import styled from '@emotion/styled'
import gql from 'graphql-tag'
import { Taxonomy } from '../../generated/myskills'
import { useMutation } from 'react-apollo-hooks'

const List = styled.ul`
  color: white;
  font-family: 'Arial';
  padding: 0;
  font-weight: bold;
  text-align: center;
  li {
    margin-bottom: 5px;
    list-style: none;
    transition: all 100ms ease;

    &:hover {
      transform: scale(1.1);
    }
  }
`

export const ADD_EXPERIENCE_CLIENT = gql`
  mutation addExperience($experience: ExperienceInput!) {
    addExperience(experience: $experience) @client
  }
`

interface AddExperienceListProps {
  experiences: Taxonomy.Result[]
}

const AddExperienceList: React.FC<AddExperienceListProps> = ({
  experiences,
}) => {
  const addExperience = useMutation(ADD_EXPERIENCE_CLIENT)

  return (
    <List>
      {experiences.map((experience: Taxonomy.Result, i: number) => {
        return (
          <li key={i}>
            <button
              onClick={() =>
                addExperience({
                  variables: {
                    experience: {
                      name: experience.term,
                      years: '',
                      taxonomyId: experience.taxonomyId,
                    },
                  },
                })
              }
            >
              {experience.term}
            </button>
          </li>
        )
      })}
    </List>
  )
}

export default AddExperienceList
