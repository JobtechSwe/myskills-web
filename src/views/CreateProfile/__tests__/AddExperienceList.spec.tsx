import * as React from 'react'
import AddExperienceList from '../AddExperienceList'
import { render } from '../../../utils/test-utils'
import { wait, fireEvent } from 'react-testing-library'

describe('views/AddExperienceList', () => {
  let experiences: any

  beforeEach(() => {
    experiences = [
      {
        term: 'Javascript',
        taxonomyId: 'abc',
      },
      {
        term: 'Java',
        taxonomyId: 'cda',
      },
    ]
  })

  it('renders empty list', async () => {
    const { container } = render(<AddExperienceList experiences={[]} />)

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('renders experineces list', async () => {
    const { container } = render(
      <AddExperienceList experiences={experiences} />
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
