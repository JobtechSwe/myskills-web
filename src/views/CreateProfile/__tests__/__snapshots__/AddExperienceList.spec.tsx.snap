// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/AddExperienceList renders empty list 1`] = `
<div>
  <ul
    class="css-ut905x"
  />
</div>
`;

exports[`views/AddExperienceList renders experineces list 1`] = `
<div>
  <ul
    class="css-ut905x"
  >
    <li>
      <button>
        Javascript
      </button>
    </li>
    <li>
      <button>
        Java
      </button>
    </li>
  </ul>
</div>
`;
